hooks/useZoraPremint.tsx

```diff
@@ -15,44 +15,95 @@ const useZoraPremint = () => {
+      "0x9d8778a3ec8c707da2db94e35372c92afb6d39d87a703745cb3742a6d1bf479b",
+      "0xd21f5feddcc3d35cd8be74a8fab3d273bcf105802e994e6cd38148a3f0bb409b",
+    ],
+    "0x254768D47Cf8958a68242ce5AA1aDB401E1feF2B": [],
+    "0xA0EA34448738357e0c2e58147b5719A19022ac76": [
+      "0x0ba6f3024273ff5ad6a1f2b31edc007340a4506dac805115ba4a6357d03cfdb6",
+      "0xb963774743ecd20c08006daac20e62eddd7ce74623fa596939b0a316fa9f5dd6",
+      "0xf71a76852c04934569812e34ed297ac538e514d698a7ceb323a478c7fa743b48",
+    ],
+  }
+
   const mint = async (quantity = 1) => {
     try {
       if (!externalWallet?.address) {
-        await connectWallet()
-        return { error: true }
+        const connectionResult: any = await connectWallet()
+        if ((connectionResult as { error: any })?.error) {
+          toast.error("Failed to connect wallet")
+          return { error: true }
+        }
       }
 
-      const zoraFee = (await getZoraFee(quantity)) as any
+      const zoraFee = await getZoraFee(quantity)
       const presaleMaxQuantity = BigInt(1000)
       const pricePerToken = BigInt(0)
-      const sweetman = "0xcfBf34d385EA2d5Eb947063b67eA226dcDA3DC38".toLowerCase()
-      const isSweetman = externalWallet.address.toLowerCase() === sweetman
-      const SWEETMAN_PROOF = "0x79ae2e7ba528b36d9095a66991f47b7f3ac877be97dda6d556bc14baa58fd48c"
-      const DACIRKUS_PROOF = "0xc747f46b8f122cd0590c39817a019cbc336648ee109c705a235361b3ae9089df"
-      const merkleProof = isSweetman ? SWEETMAN_PROOF : DACIRKUS_PROOF
-      const args = [quantity, presaleMaxQuantity, pricePerToken, [merkleProof], COMMENT]
+      const userProof = proofs[externalWallet.address]
+      console.log(externalWallet.address)
+      if (!userProof || userProof.length === 0) {
+        toast.error("No proof found for address")
+        return { error: true }
+      }
+
+      const args = [quantity, presaleMaxQuantity, pricePerToken, userProof, COMMENT]
       const price = zoraFee[1] * BigInt(quantity)
       const totalPrice = price.toString()
       const hexValue = numberToHex(BigInt(totalPrice))
 
-      const hash = await sendTxByWallet(
-        DROP_ADDRESS,
-        CHAIN_ID,
-        abi,
-        "purchasePresaleWithComment",
-        args,
-        hexValue,
-      )
+      let methodName = "purchasePresaleWithComment"
+      if (IS_TESTNET) {
+        methodName = "purchasePresaleWithRewards"
+        args.push(zeroAddress)
+      }
+
+      const hash = await sendTxByWallet(DROP_ADDRESS, CHAIN_ID, abi, methodName, args, hexValue)
+
       if ((hash as { error: any })?.error) {
+        toast.error("Rejected transaction")
         return { error: true }
       }
+
       const response = await publicClient.waitForTransactionReceipt({
         hash: hash as Address,
       })
+
       toast.success("Collected!")
       return getTokenId(response)
     } catch (err) {
       handleTxError(err)
+      toast.error("Transaction rejected")
```

Comment:
toast.error already happens in `handleTxError` right?
- why toast a second time?

Reply:
Added this comment changes to the PR.